libraries/botbuilder-dialogs-adaptive/src/actions/httpRequest.ts

```diff
@@ -286,88 +287,110 @@ export class HttpRequest<O extends object = {}> extends Dialog<O> implements Htt
+            }
 
-        const result = new Result(response.headers);
-        result.statusCode = response.status;
-        result.reasonPhrase = response.statusText;
-
-        switch (this.responseType.getValue(dc.state)) {
-            case ResponsesTypes.Activity:
-                result.content = await response.json();
-                dc.context.sendActivity(result.content as Activity);
-                break;
-            case ResponsesTypes.Activities:
-                result.content = await response.json();
-                dc.context.sendActivities(result.content as Activity[]);
-                break;
-            case ResponsesTypes.Json: {
-                const content = await response.text();
-                try {
-                    result.content = JSON.parse(content);
-                } catch {
-                    result.content = content;
+            const result = new Result(response.headers);
+            result.statusCode = response.status;
+            result.reasonPhrase = response.statusText;
+
+            switch (this.responseType.getValue(dc.state)) {
+                case ResponsesTypes.Activity:
+                    result.content = await response.json();
+                    dc.context.sendActivity(result.content as Activity);
+                    break;
+                case ResponsesTypes.Activities:
+                    result.content = await response.json();
+                    dc.context.sendActivities(result.content as Activity[]);
+                    break;
+                case ResponsesTypes.Json: {
+                    const content = await response.text();
+                    try {
+                        result.content = JSON.parse(content);
+                    } catch {
+                        result.content = content;
+                    }
+                    break;
                 }
-                break;
+                case ResponsesTypes.Binary: {
+                    const buffer = await response.arrayBuffer();
+                    result.content = new Uint8Array(buffer);
+                    break;
+                }
+                case ResponsesTypes.None:
+                default:
+                    break;
             }
-            case ResponsesTypes.Binary: {
-                const buffer = await response.arrayBuffer();
-                result.content = new Uint8Array(buffer);
-                break;
+
+            traceInfo.response = result;
+
+            // Write trace activity for http request and response values.
+            await dc.context.sendTraceActivity('HttpRequest', traceInfo, 'Microsoft.HttpRequest', this.id);
+
+            if (this.resultProperty) {
+                dc.state.setValue(this.resultProperty.getValue(dc.state), result);
             }
-            case ResponsesTypes.None:
-            default:
-                break;
-        }
 
-        traceInfo.response = result;
+            return await dc.endDialog(result);
```

Comment:
We should put these functionality into another method like Eric's PR, please look at the changes

Reply:
It should be in a function, so iits used here and in the catch